client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx

```diff
@@ -90,19 +79,18 @@ const PlansWrapper: React.FC< Props > = ( props ) => {
 
 	const site = useSite();
 	const { __ } = useI18n();
-
+	const translate = useTranslate();
 	const isDesktop = useDesktopBreakpoint();
 	const stepName = 'plans';
 	const isReskinned = true;
 	const customerType = 'personal';
 	const isInVerticalScrollingPlansExperiment = true;
-	const planTypes = getPlanTypes( props?.flowName, reduxHideFreePlan, props.hostingFlow );
 	const headerText = __( 'Choose a plan' );
 	const isInSignup = props?.flowName === DOMAIN_UPSELL_FLOW ? false : true;
-
-	const hideFreePlan = planTypes ? ! planTypes.includes( TYPE_FREE ) : reduxHideFreePlan;
-
-	const translate = useTranslate();
+	const intent = getIntent( props?.flowName, props.hostingFlow );
+	const hideFreePlan = intent
+		? reduxHideFreePlan && 'blog-onboarding' === intent
+		: reduxHideFreePlan;
```

Comment:
> For example, the LIB flow sets the value to true [here](https://github.com/Automattic/wp-calypso/blob/dfcaffcac16c408f3be4fd9f38ef76f80ff1ddf3/client/landing/stepper/declarative-flow/link-in-bio-domain.ts#L121), but it will have no effect.

Yes, good catch. `hideFreePlan` should pass through as `reduxHideFreePlan`, although that's also a property I want to deprecate next (all of the "hideXYZPlan` should go as they are largely unnecessary to be propagating throughout). I will revisit and see how to incorporate the value from `reduxHideFreePlan` into the intent hook.

Reply:
In any case, I worked out the change in https://github.com/Automattic/wp-calypso/tree/8ce104d4f3d9bd4b029eb632a89d4ea1040ef7af to respect store flag unconditionally (and how I'd approach getting rid of props drilling along the way), and reverted in https://github.com/Automattic/wp-calypso/pull/78428/commits/7309fd36e3b7971923f235e5121bd5f178902693 as that deviates from production.

If you have a spare cycle, then maybe confirm with me if you are seeing the same.